src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs

```diff
@@ -778,10 +777,10 @@ private bool TryAddInternal(TKey key, int hashcode, TValue value, bool updateIfE
                     }
 
                     // Try to find this key in the bucket
-                    Node prev = null;
```

Comment:
it would be ideal if we could express: TKey cannot be null

Reply:
Maybe we should also add a where TKey : object constraint to this class?

Reply:
> Maybe we should also add a where TKey : object constraint to this class?

Yes